service/service_webhook_test.go

```diff
@@ -0,0 +1,189 @@
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/mitchellh/mapstructure"
+	v1client "github.com/rancher/go-rancher/client"
+	"github.com/rancher/go-rancher/v2"
+
+	"github.com/rancher/webhook-service/drivers"
+	"github.com/rancher/webhook-service/model"
+)
+
+func TestCreateAndUpgdate(t *testing.T) {
+	// Test creating a webhook
+	constructURL := fmt.Sprintf("%s/v1-webhooks/receivers?projectId=1a1", server.URL)
+	jsonStr := []byte(`{"driver":"serviceWebhook","name":"wh-name",
+		"serviceWebhookConfig": {"serviceURL": "http"}}`)
+	request, err := http.NewRequest("POST", constructURL, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		t.Fatal(err)
+	}
+	request.Header.Set("Content-Type", "application/json")
+	response := httptest.NewRecorder()
+	handler := HandleError(schemas, r.ConstructPayload)
+	handler.ServeHTTP(response, request)
+	if response.Code != 200 {
+		t.Fatalf("StatusCode %d means ConstructPayloadTest failed", response.Code)
```

Comment:
It's unit test. You don't need error messages here. The test case should make it clear, or you can add comments